modules/holodeckb2b-core/src/test/java/org/holodeckb2b/security/handlers/RaiseSignatureCreatedEventTest.java

```diff
@@ -1,18 +1,209 @@
+        }
+
+        // Invoking CreateSecurityHeaders handler
+        try {
+            Handler.InvocationResponse invokeResp = wssHeadersHandler.invoke(mc);
+            assertEquals(Handler.InvocationResponse.CONTINUE, invokeResp);
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        PMode pmode = new PMode();
+        pmode.setMep(EbMSConstants.ONE_WAY_MEP);
+        pmode.setMepBinding(EbMSConstants.ONE_WAY_PUSH);
+
+        // Setting security configuration
+        SecurityConfig secConfig = new SecurityConfig();
+        secConfig.setSignatureConfiguration(sigConfig);
+
+        PartnerConfig initiator = new PartnerConfig();
+        initiator.setSecurityConfiguration(secConfig);
+        pmode.setInitiator(initiator);
+
+        Leg leg = new Leg();
+
+        Protocol protocolConfig = new Protocol();
+        protocolConfig.setAddress("address");
+        protocolConfig.setAddActorOrRoleAttribute(true);
+
+        leg.setProtocol(protocolConfig);
+        pmode.addLeg(leg);
+
+        OMElement agreementRef =
+                AgreementRef.getElement(CollaborationInfo.getElement(userMessage));
+        String pmodeId = agreementRef.getText();
+
+        pmode.setId(pmodeId);
+
+        //Adding PMode to the managed PMode set.
+        core.getPModeSet().add(pmode);
+
+        EntityProxy<org.holodeckb2b.ebms3.persistency.entities.UserMessage>
+                userMessageEntityProxy = null;
+        try {
+            userMessageEntityProxy =
+                    MessageUnitDAO.storeReceivedMessageUnit(
+                            UserMessage.readElement(userMessage));
+        } catch (PackagingException e) {
+            fail(e.getMessage());
+        }
+        userMessageEntityProxy.entity.setPMode(pmode.getId());
+        mc.setProperty(MessageContextProperties.OUT_USER_MESSAGE,
+                userMessageEntityProxy);
+
+        // Adding event processor to make sure the SignatureCreatedEvent
+        // is actually raised.
+        final TestEventProcessor eventProcessor = new TestEventProcessor();
+        core.setEventProcessor(eventProcessor);
+
+        // Invoking RaiseSignatureEvent handler
+        try {
+            Handler.InvocationResponse invokeResp = handler.invoke(mc);
+            assertEquals(Handler.InvocationResponse.CONTINUE, invokeResp);
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        assertTrue(eventProcessor.allPurgeEvents);
+        assertEquals(1, eventProcessor.msgIdsPurged.size());
+
+        String msgId = null;
+        OMElement miElement = MessageInfo.getElement(userMessage);
+        Iterator it = miElement.getChildrenWithName(MESSAGE_ID_ELEMENT_NAME);
+        if(it.hasNext()) {
+            msgId = ((OMElement)it.next()).getText();
+        }
+        assertEquals(msgId, eventProcessor.msgIdsPurged.get(0));
+    }
+
+    class TestEventProcessor implements IMessageProcessingEventProcessor {
```

Comment:
Maybe it is more convenient to have the TestEventProcessor as a generic test utility, i.e. as a separate class? 
If it just registers all events triggered (for example by adding them to a list) the test class can always check if he correct events were triggered.